README.md

```diff
@@ -132,10 +132,109 @@ is not supported anymore. Exporting `$ROLENAME_provider_os_default` allows the u
 in Ansible) and thus get a consistent setting for all the systems of the given OS version without
 having to decide what the actual value is - the decision is delegated to the role.)
 
-
 Implementation considerations
 ===========
 
+## Role Structure
+Avoid testing for distribution and version in tasks. Rather add a variable file to "vars/"
+for each supported distribution and version with the variables that need to change according
+to the distribution and version. This way it is easy to add support to a new distribution by
+simply dropping a new file in to "vars/", see below
+[Supporting multiple distributions and versions](#supporting-multiple-distributions-and-versions). See also
+[Vars vs Defaults](#vars-vs-defaults) which mandates "Avoid embedding large lists or "magic values" directly
+into the playbook." Since distribution-specific values are kind of "magic values", it applies to them. The
+same logic applies for providers: a role can load a provider-specific variable file, include a
+provider-specific task file, or both, as needed. Consider making paths to templates internal variables if you
+need different templates for different distributions.
+
+## Check Mode and Idempotency Issues
+
+* The role should work in check mode, meaning that first of all, they should not fail check mode, and
+  they should also not report changes when there are no changes to be done. If it is not possible
+  to support it, please state the fact and provide justification in the documentation.
+  This applies to the first run of the role.
+
+* Reporting changes properly is related to the other requirement: **idempotency**. Roles
+  should not perform changes when applied a second time to the same system with the same parameters,
+  and it should not report that changes have been done if they have not been done. Due to this,
+  using `command:` is problematic, as it always reports changes. Therefore, override the result by
+  using `changed_when:`
```

Comment:
Should we add a note here like "In order to support idempotency, you may need to check the result of the command.  If so, use `register: rolename_register_somevarname` and add another task to check the result (return code, output) to report if the previous `command` changed the state or not." ?

Reply:
I don't think so, `change_when` should take care of it, no? Your suggestion would be used only if determining whether there is a change can not be constructed as a Jinja2 expression, which should be quite a rare case.